### PR TITLE
HIP fixes for CEED benchmarks

### PR DESCRIPTION
**Pull request by camierjs**
Small fixes to be able to run the CEED benchmarks:
- ~~**makefile**: MFEM_CONFIG_FILE needs to be escaped~~
- **linalg/vector.cpp**: similar `hipVectorMin` / `hipVectorDot` than the CUDA ones

| PR | Author | Editor | Reviewers  | Assignment | Approval | Merge |
| --- | --- | --- | ---  | --- | --- | --- |
| https://github.com/mfem/mfem/pull/1198 | @camierjs| @tzanio | @dmed256 + @acfisher | 12/18/19 | 12/28/19 | ⌛due 01/04/20 |

### CODE REVIEW

File: linalg/vector.cpp [hunk 1]
```diff
@@ -932,12 +932,100 @@ static double cuVectorDot(const int N, const double *X, const double *Y)
...
+
+static double hipVectorMin(const int N, const double *X)
+{
+   const int tpb = MFEM_CUDA_BLOCKS;
+   const int blockSize = MFEM_CUDA_BLOCKS;
+   const int gridSize = (N+blockSize-1)/blockSize;
+   const int min_sz = (N%tpb)==0? (N/tpb) : (1+N/tpb);
+   cuda_reduce_buf.SetSize(min_sz);
+   Memory<double> &buf = cuda_reduce_buf.GetMemory();
+   double *d_min = buf.Write(MemoryClass::CUDA, min_sz);
```
Comment (artv3) [inline linalg/vector.cpp]: Should the memory class be CUDA?

Comment (camierjs) [reply linalg/vector.cpp]: It's still CUDA, in the current memory manager, we don't really have a HIP backend and we are using the CUDA enum as some 'DEVICE' one. All the cuda* are just forwarded to the hip* ones.

File: makefile [hunk 2]
```diff
@@ -137,7 +137,7 @@ BUILD_DIR := $(MFEM_BUILD_DIR)
 BUILD_REAL_DIR := $(abspath $(BUILD_DIR))
 ifneq ($(BUILD_REAL_DIR),$(MFEM_REAL_DIR))
    BUILD_SUBDIRS = $(DIRS) config $(EM_DIRS) doc $(TEST_DIRS)
-   CONFIG_FILE_DEF = -DMFEM_CONFIG_FILE='"$(BUILD_REAL_DIR)/config/_config.hpp"'
+   CONFIG_FILE_DEF = -DMFEM_CONFIG_FILE='\"$(BUILD_REAL_DIR)/config/_config.hpp\"'
```
Comment (v-dobrev) [inline makefile]: This seems to break the out-of-source build for me, e.g. I get errors when I try this:
```console
cd mfem
mkdir build
cd build
make -f ../makefile config
make
```

Comment (camierjs) [reply makefile]: You're right, I'll look on the CEED benchmark side why I needed that there.

Comment (acfisher) [reply makefile]: Did this get taken care of?

Comment (tzanio) [reply makefile]: Yes, see https://github.com/mfem/mfem/pull/1198/commits/fde20508f165cbb256fd9ce752229909b8572c4f

**Comment by tzanio** [comment]
@dmed256, can you take a quick look?

File: linalg/vector.cpp [hunk 3]
```diff
@@ -932,12 +932,100 @@ static double cuVectorDot(const int N, const double *X, const double *Y)
...
+   Memory<double> &buf = cuda_reduce_buf.GetMemory();
+   double *d_dot = buf.Write(MemoryClass::CUDA, dot_sz);
+   hipLaunchKernelGGL(hipKernelDot,gridSize,blockSize,0,0,N,d_dot,X,Y);
+   MFEM_GPU_CHECK(hipGetLastError());
+   const double *h_dot = buf.Read(MemoryClass::HOST, dot_sz);
+   double dot = 0.0;
+   for (int i = 0; i < dot_sz; i++) { dot += h_dot[i]; }
+   return dot;
+}
+#endif // MFEM_USE_CUDA
```
Comment (dmed256) [inline linalg/vector.cpp]:
```suggestion
#endif // MFEM_USE_HIP
```
^ super minor thing

**Comment by tzanio** [comment]
Thanks @camierjs!

**Pull request merged**